client/state/reader/streams/reducer.js

```diff
@@ -122,6 +123,13 @@ export const items = ( state = [], action ) => {
 
 			// Filter out duplicate x-posts
 			return combineXPosts( newState );
+		case READER_STREAMS_NEW_POST_RECEIVE: {
+			const newPost = {
+				...action.payload.postData,
+				isSynthetic: true,
+			};
+			return [ newPost, ...state ];
+		}
```

Comment:
Rather than use this `isSynthetic` flag, what I want to do is introduce a new object called `temporaryItems` or similar in the `posts` store, which we would populate with the API response, and use that in conjunction with the regular items property from the `posts` store when rendering the Stream component.

Reply:
Makes sense.